internal/promqle2etest/backend_otelgcm.go

```diff
@@ -0,0 +1,196 @@
+	ports := map[string]int{"http": 9090}
+
+	f := env.Runnable(name).WithPorts(ports).Future()
+
+	// NOTE(bwplotka): Starting from https://cloud.google.com/stackdriver/docs/managed-prometheus/setup-otel#run-off-gcp
+	// but with extra things to make this actually work (TODO: update the docs)
+	// * report_extra_scrape_metrics for compatibility.
+	// * Telemetry in Prometheus format.
+	// * GCM exporter: extra_metrics_config: enable_target_info: false and enable_scope_info: false
+	// * Also: feature gate for native histograms in flags
+	config := fmt.Sprintf(`
+receivers:
+  prometheus:
+    report_extra_scrape_metrics: true
+    config:
+      scrape_configs:
+      - job_name: 'test'
+        scrape_interval: 5s
+        scrape_timeout: 5s
+        static_configs:
+        - targets: [%s]
+        metric_relabel_configs:
+        - regex: instance
+          action: labeldrop
+
+processors:
+  resource:
+    attributes:
+    - key: "cluster"
+      value: "%s"
+      action: upsert
+    - key: "location"
+      value: "%s"
+      action: upsert
+    - key: "collector"
+      value: "%s"
+      action: upsert
+
+  transform:
+    # "location", "cluster", "namespace", "job", "instance", and "project_id" are reserved, and
+    # metrics containing these labels will be rejected. Prefix them with exported_ to prevent this.
+    # TODO(bwplotka): Update this and docs. Below gives warning 13:02:20 otel-gcm: 2025-04-14T12:02:20.243Z	info	ottl@v0.123.0/parser_collection.go:447	one or more paths were modified to include their context prefix, please rewrite them accordingly.
+    metric_statements:
+    - context: datapoint
+      statements:
+      - set(attributes["exported_location"], attributes["location"])
+      - delete_key(attributes, "location")
+      - set(attributes["exported_cluster"], attributes["cluster"])
+      - delete_key(attributes, "cluster")
+      - set(attributes["exported_namespace"], attributes["namespace"])
+      - delete_key(attributes, "namespace")
+      - set(attributes["exported_job"], attributes["job"])
+      - delete_key(attributes, "job")
+      - set(attributes["exported_instance"], attributes["instance"])
+      - delete_key(attributes, "instance")
+      - set(attributes["exported_project_id"], attributes["project_id"])
+      - delete_key(attributes, "project_id")
+
+  batch:
+    # batch metrics before sending to reduce API usage
+    send_batch_max_size: 200
+    send_batch_size: 200
+    timeout: 5s
+
+  memory_limiter:
+    # drop metrics if memory usage gets too high
+    check_interval: 1s
+    limit_percentage: 65
+    spike_limit_percentage: 20
+
+# Note that the googlemanagedprometheus exporter block is intentionally blank
+exporters:
+  googlemanagedprometheus:
+    project: "%s"
+    # TODO(bwplotka): Do we need to disable cumulative_normalization explicitly?
```

Comment:
We should allow the test to pass in a config imo. We want to test using the `googlemanagedprometheus` exporter but we also want to test that the `otlp` exporter works the same way when used with the `metricstarttime` processor

Reply:
Yea, totally, I just want to make sure GMP path works too, something timeouts.

Reply:
Do you have a container image that is built on top of your PR I could use?  https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/38594